hazelcast/src/main/java/com/hazelcast/topic/MessageListener.java

```diff
@@ -20,23 +20,23 @@
 
 /**
  * Message listener for {@link ITopic}.
- *
- * Provided that a MessageListener is not registered twice, a MessageListener will never be called concurrently. So there
- * is no need to provide thread-safety on internal state in the MessageListener. Also there is no need to enforce safe
- * publication, the ITopic is responsible for the memory consistency effects. In other words, there is no need to make
+ * <p>
+ * A MessageListener will never be called concurrently (provided that it's not registered twice). So there
+ * is no need to synchronize access to the internal state in the MessageListener. Also there is no need to enforce
+ * safe publication, the ITopic is responsible for memory consistency effects. In other words, there is no need to make
  * internal fields of the MessageListener volatile or access them using synchronized blocks.
  *
- * @param <E> message
+ * @param <E> message type
  */
 @FunctionalInterface
 public interface MessageListener<E> extends EventListener {
```

Comment:
Changes to `ReplicatedMap`, `NamedOperation`, `BaseInvocation`, and `MessageListener` are not relevant to this PR and the SQL engine as a whole. I would propose to apply them in a separate PR to avoid mixing changes to unrelated parts of the product in a single PR.